com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ShaderUtil.java

```diff
@@ -510,13 +534,31 @@ public static Result transform(String input, ShaderType shaderType, String targe
                                 layout = "layout(set=" + layoutSet + ")";
                             }
 
-                            if (isOpaque){
+                            if (isOpaque) {
                                 line = layout + " " + line;
                             } else {
-                                line = "\n" + layout + " " + keyword + " " + glUBRep + ubIndex++ + " { " +
+                                line = "\n" + layout + " " + keyword + " " + ubBase + ubIndex++ + " { " +
                                 (precision == null ? "" : (precision + " ")) + type + " " + identifier + " " + (any == null ? "" : (any + " ")) + "; };";
                             }
                         }
+                    } else {
+                        Matcher uniforBlockBeginMatcher = Common.regexUniformBlockBeginKeywordPattern.matcher(line);
+                        if (uniforBlockBeginMatcher.find()) {
+                            String keyword = uniforBlockBeginMatcher.group("keyword");
+                            if(keyword != null) {
+                                String layout = uniforBlockBeginMatcher.group("layout");
+                                String type   = uniforBlockBeginMatcher.group("type");
+                                String any    = uniforBlockBeginMatcher.group("any");
+
+                                boolean blockScopeBegin = line.contains("{");
+
+                                if (layout == null) {
+                                    layout = "layout(set=" + layoutSet + ")";
+                                }
+
+                                line = layout + " " + keyword + " " + type + (blockScopeBegin ? "{" : "");
```

Comment:
The difference between this and the line above is that this handles uniform buffers and the above handles uniforms that are *NOT* in a buffer. The difference is more or less that in this case we don't match the ending ';' character, and we don't care about the identifier (which we might want to do later with pure GLSL).